xbmc/network/websocket/WebSocket.cpp

```diff
@@ -85,12 +86,15 @@ CWebSocketFrame::CWebSocketFrame(const char* data, uint64_t length)
   int offset = 0;
   if (m_length == 126)
   {
-    m_length = (uint64_t)Endian_SwapBE16(*(const uint16_t *)(m_data + 2));
+    uint16_t length;
+    std::memcpy(&length, m_data + 2, 2);
```

Comment:
```suggestion
    offset = 2;
    std::memcpy(&length, m_data + 2, offset);
```
same below with offset 8?